src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -67,7 +67,14 @@ protected ApiConfiguration ApiConfiguration
                         var builder = this.ConfigureApi(new ApiBuilder());
                         EnableConventions(builder, apiType);
                         ApiConfiguratorAttribute.ApplyApiBuilder(apiType, builder);
-                        builder.TryUseSharedApiScope(); // TODO: Maybe default to context scope?
+                        if (!builder.HasService<IApi>())
+                        {
+                            builder.Services
+                                .AddScoped(apiType, sp => sp.GetService<ApiScope>().Api)
```

Comment:
Shall we just use `GetService<IApi>` so here you don't have to register the service for `apiType`?